### DISCUSSION TITLE
Multipurpose server

### DISCUSSION THREAD

**Discussion by gweobia** [strongswan/strongswan#1082] [Q&A]
I am trying to write **configuration** for a multipurpose server that should accept connections from every client that tries to connect to it. It should be **maximum friendly** to all implementations of VPN clients in all major desktop, server, mobile and embedded device OSes.

As far as I understand the key idea is to negotiate with peer to use any (or first?) combination of authentication and encryption algorithms client supports (except no encryption). Is the **solution** to list all variations (Cartesian product?) of the algorithms of IPsec specification in **proposals** property of swanctl.conf in the connection descriptor?

Maybe there are **more ideas** I should keep in mind to achieve the goal?

**Comment by tobiasbrunner** [comment]
Look at [the docs](https://docs.strongswan.org/docs/5.9/index.html). There are example configs and notes on interoperability.
